gcc/rust/hir/rust-ast-lower-type.h

```diff
@@ -62,20 +62,18 @@ class ASTLowerTypePath : public ASTLoweringBase
   {
     std::vector<std::unique_ptr<HIR::TypePathSegment>> translated_segments;
 
-    path.iterate_segments ([&] (AST::TypePathSegment *seg) mutable -> bool {
-      translated_segment = nullptr;
-      seg->accept_vis (*this);
-      if (translated_segment == nullptr)
-	{
-	  rust_fatal_error (seg->get_locus (),
-			    "failed to translate AST TypePathSegment");
-	  return false;
-	}
-
-      translated_segments.push_back (
-	std::unique_ptr<HIR::TypePathSegment> (translated_segment));
-      return true;
-    });
+    for (auto &seg : path.get_segments ())
+      {
+	translated_segment = nullptr;
+	seg->accept_vis (*this);
+	if (translated_segment == nullptr)
+	  {
+	    rust_fatal_error (seg->get_locus (),
```

Comment:
This is not related to your PR but lets raise an issue about this fatal_error it should be an assertion

Reply:
Thanks for you quick review!

Are you saying I should replace `rust_fatal_error` with another function or another procedure?

Reply:
Not as part of this pr. I raised a separate issue for that.